charts/timescaledb-single/scripts/prestop.sql

```diff
@@ -60,3 +60,7 @@ CHECKPOINT;
     SELECT 'We are a primary: Successfully invoked checkpoints, now issuing a switchover.';
     \! curl -s http://localhost:8008/switchover -XPOST -d '{"leader": "$(hostname)"}'
 \endif
+
+\if :linkerdShutdown
+    \! curl -s -m 5 -X POST http://localhost:4191/shutdown
```

Comment:
Will the linkerd shutdown disconnect the pod from the outside connections, in particular, S3 and replicas? We still need to archive the shutdown checkpoint, so it is only safe to cut the pod from the network when PostgreSQL is fully stopped.

Reply:
CHECKPOINT should not return until that has been performed, as long as archival is configured. Else, is it even safe to call `/switchover` a few lines above?

Reply:
Well, Patroni would/should handle ensuring that it is safe to perform a switchover; so I suppose the primary concern is about ensuring that the checkpoint is archived.

I'll add a line to check `SELECT * FROM pg_stat_archiver;` to ensure that we everything is finished before we issue the linkerd shutdown.

Reply:
Ok, posting a quick follow-up here. The call to /shutdown terminates active connections pretty much immediately. And it does NOT appear as though `CHECKPOINT` does any waiting at all for an archival process to finish.

I'm thinking that our best bet right now is likely to check the status of the archiver: `pgbackrest info --output=json | jq -r '.[0].archive'`. Maybe cross-referencing that info with pg_stat_archiver and such.

Reply:
Ok, I think there is actually a much more simple solution here. Let me put the commit together real quick.

Reply:
@alexeyklyukin I'm going to close this thread, as there are no longer changes in this specific location. When you have the time, please let me know what you think of the new solution.